programs/ssl/ssl_client2.c

```diff
@@ -2995,6 +3007,25 @@ int main(int argc, char *argv[])
                            (unsigned int) -ret);
             goto exit;
         }
+#if defined(MBEDTLS_SSL_PROTO_TLS1_3) && defined(MBEDTLS_SSL_EARLY_DATA)
+        if (opt.early_data == MBEDTLS_SSL_EARLY_DATA_ENABLED
+            && strlen(opt.early_data_file) > 0) {
```

Comment:
I do not think this limitation match 2nd way from Ronald
`2 .two ssl_client2 options, one to enable and the other to pass the path of the file containing the data to transfer. In that case no file or an empty file are equivalent.`

As my understand, 
- It does not means we should not open early data file when disable. 
- `empty file` does not means file name len is zero. You should check the file length. 

On my view, this check make things complex.

Reply:
If the two options are at their default value then better to not do anything I'd say.
Then if early_data == MBEDTLS_SSL_EARLY_DATA_ENABLED call write_early_data in all cases (with a 0 length if no file defined or an empty file.
If early_data == MBEDTLS_SSL_EARLY_DATA_DISABLED call also write_early_data in all cases (with a 0 length if the file is empty) and expect the function to return in error.

Reply:
understand.